### PR TITLE
[REVIEW] cuio: factor out parse_options creation to pure function

### PR DESCRIPTION
**Pull request by cwharris**
This PR factors out the `parse_options` creation in the csv reader constructor in to it's own pure function. It also renames `parseOptions` => `parse_options`, upgrades `parse_options` to an owning type (for the trie nodes), and creates a `parse_options_view` as a non-owning type which can be used on-device.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by davidwendt** [comment]
`branch-0.16` is in burn down mode. This should probably move to `branch-0.17`.
https://docs.rapids.ai/releases/process/#burn-down

**Comment by cwharris** [comment]
@davidwendt My bad. Habit. Updated.

**Comment by cwharris** [comment]
rerun tests

**Comment by cwharris** [comment]
rerun tests

**Comment by cwharris** [comment]
rerun tests

**Comment by cwharris** [comment]
rerun tests

File: cpp/src/io/utilities/parsing_utils.cuh [hunk 1]
```diff
@@ -36,12 +41,47 @@ struct ParseOptions {
...
 };
 
+struct parse_options {
+  char delimiter;
+  char terminator;
+  char quotechar;
+  char decimal;
+  char thousands;
+  char comment;
+  bool keepquotes;
```
Comment (davidwendt) [inline cpp/src/io/utilities/parsing_utils.cuh:57]: Should the multi-word members be snake case?
```
keep_quotes
double_quote
day_first
skip_blank_lines
```

**Review by davidwendt — APPROVED** [review]
Looks good to me.
Perhaps a cuio person should review this.

**Review by rgsl888prabhu — APPROVED** [review]
LGTM

File: cpp/src/io/utilities/parsing_utils.cuh (hunk 1 above)
Comment (cwharris) [reply cpp/src/io/utilities/parsing_utils.cuh:57]: Probably :shrug:. We can get this cleaned up in a later PR, though. There will be quite a few.

Comment (davidwendt) [reply cpp/src/io/utilities/parsing_utils.cuh:57]: I believe these only show up in the files that are already in this PR and you are modifying all their references already.

**Pull request merged**